tests/unit/src/main/scala/tests/TestingServer.scala

```diff
@@ -473,6 +473,8 @@ final class TestingServer(
     server.executeCommand(new ExecuteCommandParams(command, args)).asScala
   }
 
+  def waitFor(sec: Long): Future[Unit] = Future { Thread.sleep(sec) }
```

Comment:
Could we maybe wait on the indexingPromise in MetalsLanguageServer? We can make it public and it should work the same.

Reply:
🤔 so I tried adding in 
```
      _ <- server.server.indexingPromise.future
```
Which I assumed would wait for it, but it seems to not work. This is what you had in mind right?

Reply:
It should wait until the workspace is indexed, so maybe we should also wait on `didSave` (it will wait on the compilation)?  The test classes are still not available by then?

Reply:
Yea but there already is a did save in the others that I tried this on. For example in `testTarget` we have the `didOpen`, `didSave`, and then the `indexingPromise.future`, and that still doesn't have the test classes. I have no idea why that takes more time.

Reply:
Damn, it might have something to do with Bloop. Alternative would be to use SBT bsp here, but it's fine to leave it as is.

Reply:
🤔 I'm thinking for now to leave it as I played around with this a bit more and still can't get it to work as expected.